src/main/java/deez/Task.java

```diff
@@ -0,0 +1,69 @@
+package deez;
+
+import java.io.Serializable;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Abstract class representing a task.
+ */
+public abstract class Task implements Serializable {
```

Comment:
If there are no abstract methods being stated, is there still a need for Task to be an abstract class?

Reply:
Hmm, there currently aren't any abstract methods because the descendants don't really share any common non-generic behaviour yet. `Task` should be an abstract class because we don't want to be able to instantiate a `Task` (which is not a `Todo`, nor a `Deadline` or an `Event`) in the main application logic.